### PR TITLE
[release/8.0-staging] Update APICompat baseline version to 8.0.0

### PR DESCRIPTION
**Pull request by ViktorHofer**
Manual backport of https://github.com/dotnet/runtime/pull/94712
Unblocks https://github.com/dotnet/runtime/pull/94193

1. Update APICompat baseline version
2. Update SDK to 8.0.100
3. Remove APICompat package reference in favor of what's in the SDK
4. Update System.Drawing.Common version to 8.0.0
5. Clear out the `ApiCompatBaseline.NetCoreAppLatestStable.xml` file as we shouldn't have any public API differences in servicing.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -383,11 +383,6 @@
       <Uri>https://github.com/dotnet/roslyn-analyzers</Uri>
       <Sha>b4d9a1334d5189172977ba8fddd00bda70161e4a</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.DotNet.ApiCompat.Task" Version="8.0.100-rtm.23520.8">
```
Comment (carlossanlop) [inline eng/Version.Details.xml:386]: Why did you decide to remove this?

File: eng/Versions.props [hunk 2]
```diff
@@ -84,8 +84,6 @@
   </PropertyGroup>
   <PropertyGroup>
     <StaticCsVersion>0.2.0</StaticCsVersion>
-    <!-- SDK dependencies -->
-    <MicrosoftDotNetApiCompatTaskVersion>8.0.100-rtm.23520.8</MicrosoftDotNetApiCompatTaskVersion>
```
Comment (carlossanlop) [inline eng/Versions.props:88]: Why did you have to remove this?

File: src/libraries/apicompat/ApiCompat.proj [hunk 3]
```diff
@@ -44,7 +44,7 @@
 
     <!-- Restore and reference assemblies not built in runtime anymore, required for resolving type forward on both comparison sides. -->
     <ApiCompatTypeForwardDestinationPackage Include="System.Data.SqlClient" PackageVersion="$(SystemDataSqlClientVersion)" TargetFramework="netcoreapp2.1" />
-    <ApiCompatTypeForwardDestinationPackage Include="System.Drawing.Common" PackageVersion="$(SystemDrawingCommonVersion)" TargetFramework="net7.0" />
```
Comment (carlossanlop) [inline src/libraries/apicompat/ApiCompat.proj:47]: Was System.Drawing giving you failures that you were not seeing in the main PR?

File: src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.xml [hunk 4]
```diff
@@ -1,2692 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <!-- https://learn.microsoft.com/en-us/dotnet/fundamentals/package-validation/diagnostic-ids -->
 <Suppressions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
-  <Suppression>
```
Comment (carlossanlop) [inline src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.xml:4]: Why did you remove all of these? They were a lot.

File: eng/Version.Details.xml (hunk 1 above)
Comment (ViktorHofer) [reply eng/Version.Details.xml:386]: We don't need live toolset updates from the SDK anymore. We stick with the 8.0.100 version.

File: src/libraries/apicompat/ApiCompat.proj (hunk 3 above)
Comment (ViktorHofer) [reply src/libraries/apicompat/ApiCompat.proj:47]: The main PR has the same change. We update System.Drawing.Common to the 8.0.0 version for consistency.

File: src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.xml (hunk 4 above)
Comment (ViktorHofer) [reply src/libraries/apicompat/ApiCompatBaseline.NetCoreAppLatestStable.xml:4]: The baseline contained all the API differences between 7.0.0 and 8.0.0. We are now bumping the baseline version to 8.0.0 so the API diff is zero.

**Comment by ViktorHofer** [comment]
Upgrading to the 8.0.100 SDK:

```
FAILED: test/ijw/CMakeFiles/ijw.dir/ijw.cpp.obj 
  C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1437~1.328\bin\Hostx86\arm64\cl.exe  /nologo /TP -DDISABLE_CONTRACTS -DHOST_64BIT -DHOST_ARM64 -DHOST_WINDOWS -DNDEBUG -DTARGET_64BIT -DTARGET_ARM64 -DTARGET_WINDOWS -DURTBLDENV_FRIENDLY=Retail -D_FILE_OFFSET_BITS=64 -Dijw_EXPORTS  /DWIN32 /D_WINDOWS    /O2 /Ob2 /DNDEBUG -MD /Ox /nologo /W4 /WX /Oi /Oy- /Gm- /Zp8 /Gy /GS /fp:precise /FC /MP /Zm200 /Zc:strictStrings /Zc:wchar_t /Zc:inline /Zc:forScope /wd4065 /wd4100 /wd4127 /wd4131 /wd4189 /wd4200 /wd4201 /wd4206 /wd4239 /wd4245 /wd4291 /wd4310 /wd4324 /wd4366 /wd4456 /wd4457 /wd4458 /wd4459 /wd4463 /wd4505 /wd4702 /wd4706 /wd4733 /wd4815 /wd4838 /wd4918 /wd4960 /wd4961 /wd5105 /wd5205 /we4007 /we4013 /we4102 /we4551 /we4700 /we4640 /we4806 /w34092 /w34121 /w34125 /w34130 /w34132 /w34212 /w34530 /w35038 /w44177 /Zi /ZH:SHA_256 /source-charset:utf-8 /GL /we4456 /we4457 /we4458 /we4459 /wd4365 /wd4679 /wd5271 /clr:netcore /AID:\a\_work\1\s\.dotnet/packs/Microsoft.NETCore.App.Ref/8.0.0/ref/net8.0 /showIncludes /Fotest\ijw\CMakeFiles\ijw.dir\ijw.cpp.obj /Fdtest\ijw\CMakeFiles\ijw.dir\ /FS -c D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): error C3611: 'System::Numerics::INumberBase::System::IUtf8SpanFormattable.TryFormat': a sealed function cannot have a pure-specifier
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Numerics::INumberBase' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Numerics::IFloatingPointConstants' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Numerics::IFloatingPoint' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Decimal' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): error C3611: 'System::Numerics::INumberBase<TSelf>::System::IUtf8SpanFormattable.TryFormat': a sealed function cannot have a pure-specifier
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: see reference to class generic instantiation 'System::Numerics::INumberBase<TSelf>' being compiled
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Numerics::IFloatingPointConstants' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Numerics::IFloatingPoint' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Decimal' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): error C3611: 'System::Numerics::INumberBase<System::Decimal>::System::IUtf8SpanFormattable.TryFormat': a sealed function cannot have a pure-specifier
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: see reference to class generic instantiation 'System::Numerics::INumberBase<System::Decimal>' being compiled
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: see reference to class generic instantiation 'System::Numerics::IFloatingPointConstants<System::Decimal>' being compiled
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: see reference to class generic instantiation 'System::Numerics::IFloatingPoint<System::Decimal>' being compiled
  D:\a\_work\1\s\src\native\corehost\test\ijw\ijw.cpp(22): note: This diagnostic occurred while importing type 'System::Decimal' from assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.
  [187/206] Building CXX object comhost\CMakeFiles\comhost.dir\__\hostmisc\longfile.windows.cpp.obj
  [188/206] Linking CXX executable test\nativehost\nativehost.exe
  [189/206] Building CXX object comhost\CMakeFiles\comhost.dir\__\hostmisc\pal.windows.cpp.obj
  [190/206] Building CXX object ijwhost\CMakeFiles\ijwhost.dir\ijwhost.cpp.obj
  [191/206] Building CXX object ijwhost\CMakeFiles\ijwhost.dir\ijwthunk.cpp.obj
  ninja: build stopped: subcommand failed.
```

cc @tannergooding @jkoritzinsky @agocke

**Comment by ViktorHofer** [comment]
Servicing approved as necessary to ship 8.0.1.

**Review by akoeplinger — APPROVED** [review]
